### ISSUE TITLE
When can we expect to use GPUs for training on Macs

### ISSUE THREAD

**Issue by Yuhuadi** [RVC-Boss/GPT-SoVITS#833]
> Note: The models trained with GPUs on Macs result in significantly lower quality compared to those trained on other devices, so we are temporarily using CPUs instead.

When can we expect to use GPUs for training on Macs?

**Comment by XXXXRT666** [comment]
This depends on PyTorch's optimization for MPS.

**Comment by Lion-Wu** [comment]
> This depends on PyTorch's optimization for MPS.

right

**Comment by thom802311** [comment]
mac用autodl租个玩玩吧。mac本地体验很差的

**Comment by XXXXRT666** [comment]
> mac用autodl租个玩玩吧。mac本地体验很差的

建议训练上云，本地推理速度过得去

**Comment by Yuhuadi** [comment]
> mac用autodl租个玩玩吧。mac本地体验很差的

Why do you think that? Mac has the largest GPU Memory among PCs, and has good GPU performance

**Comment by XXXXRT666** [comment]
Because of the attention Performance on MPS, it's not suitable for GSV training and inference, that's why we choose CPU instead. If you do want, you could try MLX

**Comment by Lion-Wu** [comment]
> > mac用autodl租个玩玩吧。mac本地体验很差的
> 
> Why do you think that? Mac has the largest GPU Memory among PCs, and has good GPU performance

You are right, but there can be errors when using MPS (Metal Performance Shaders), such as memory leaks and poor model quality, so we can only use the CPU. Currently, using a Mac for training is not the best option. If MPS can be used without issues in the future, we will support it.

**Comment by Yuhuadi** [comment]
> > > mac用autodl租个玩玩吧。mac本地体验很差的
> > 
> > 
> > Why do you think that? Mac has the largest GPU Memory among PCs, and has good GPU performance
> 
> You are right, but there can be errors when using MPS (Metal Performance Shaders), such as memory leaks and poor model quality, so we can only use the CPU. Currently, using a Mac for training is not the best option. If MPS can be used without issues in the future, we will support it.

Could you please provide some information about the errors that occurred when using MPS? Or if it has already been posted, where can I find them? I might be able to help debug
